lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3244,6 +3252,7 @@ ParseResult FIRCircuitParser::parseModule(CircuitOp circuit,
     return failure();
 
   auto moduleTarget = (circuitTarget + "|" + name.getValue()).str();
+  targets.push_back(StringAttr::get(getContext(), moduleTarget));
```

Comment:
if I understand this correctly, this is proactively building up all the targets that get parsed, even if they don't get referenced.  This is going to build a huge set which is almost never used.

Can you add this to `FIRParser::getAnnotations` somehow?  It tries pretty hard to only triggers if there is actually an annotation.  You could make it add to your set whenever there is a hit.  That would ignore all the targets that never have an annotation on them.

Reply:
I changed this towards that direction.  `StringSet`s are passed around where needed and these are mutated by any child calls to `getAnnotations` or `getSplitAnnotations`. 

(I had started off wanting to remove annotations from the map in `getAnnotations`, but that obviously wouldn't work due to the parallelism.  I then got sidetracked with the janky original approach of recording all names.  I had incorrectly thought that having to add parameters to a lot of methods was a bad idea.  That was ill-founded.)

Thanks for the great suggestion.  This is a way better implementation now.